### DISCUSSION TITLE
Support for RAW formats

### DISCUSSION THREAD

**Discussion by vishnukvmd** [ente/ente#625] [Enhancements]
We should support previews for the following raw file formats on all platforms.
- [ ] NEF
- [ ] RAF
- [ ] DNG
- [ ] TIFF

Currently transcoding is only supported on Mac. What we should do is generate a high-res preview at the time of upload that can be rendered on all platforms (mobile included).

Sample images available @ https://albums.ente.io/?t=4edQY8Lv#JBdFEj4oYbYZgerNAsnN2qVyBS174PvZ39iUtpJYKNWQ

More samples @ https://filesamples.com/categories/image

**Comment by vfxninjaeditor** [comment]
.ORF for Olympus cameras and .RW2 for Panasonic as well

**Comment by vishnukvmd** [comment]
@vfxninjaeditor would you happen to have any sample images you could help us out with?

**Reply by RolfSievert** [reply]
[ORF_sample.zip](https://github.com/user-attachments/files/18256064/ORF_sample.zip)
Just a single sample but I have loads (all from the same camera though) if you need, let me know.

**Comment by Mike-Sundays** [comment]
Any news on this feature?

**Comment by ethankrech** [comment]
Also need CR2 and CR3

**Comment by plumber-craic** [comment]
Would be great to have thumbnails and high res previews of raw images. I'm not a node dev, but Libraw seems capable of handling the raw files, and with a cpp wrapper could be called from ts. 
https://github.com/LibRaw/LibRaw

**Comment by jbelien** [comment]
Support for RAW formats would indeed make Ente even more awesome (in my case, mostly NEF)! 🤩

**Comment by 6058aa27-6ced-4d77-952a-bf55d59351c2** [comment]
Personally, a big personal barrier to using ente photos is I like to take RAW (DNG) photos along with my JPGs on my Pixel 6, and when I try to add my library to ente there's a bunch of duplicates of the same photo in multiple formats. I like what Google Photos does, where it groups multiple pictures from the same shot under one image, including RAW, JPG, and crops. But just hiding the DNGs would also be good. I don't really use them on my phone, they're just nice to have, especially if I want to edit it in a proper program.

**Reply by randolph-** [reply]
I agree! I opened a feature request on this last summer if anyone would like to upvote: https://github.com/ente-io/ente/discussions/2276

**Comment by defconxt** [comment]
HEIC and DNG preview support?

**Comment by Zvenor** [comment]
.ARW would be great also!

**Reply by yajo** [reply]
For samples, you can download [the ARW provided here](https://discuss.pixls.us/t/would-love-to-see-how-you-handle-this-in-dt/50872). In that forum is easy to find samples for raw formats.

I wrongly submitted this issue downstream: https://github.com/NixOS/nixpkgs/issues/420929. The problem seems to be here upstream indeed. The official AppImage build also fails.

**Comment by v1nodp** [comment]
No love for .CR2 yet, even though I use it very less. But many pictures with .CR2 is just black in the app. Would love to see it being added to the list.

**Comment by abdul-mustapha** [comment]
DNGs seems to be hit and miss for me, sometimes it generates the thumbnail, sometimes it stays black.

**Comment by xavaero** [comment]
CR2 support would be nice. I shoot a lot in RAW, and Ente shows my photos in the app all black without a thumbnail.

**Comment by ghost** [comment]
Hi there! Just checking in to see if there are any updates on the CR2 and CR3 RAW support. Would love to hear any news!

**Comment by dm30** [comment]
Where can we send you samples to help you testing?

**Reply by vishnukvmd** [reply]
Please feel free to share samples right here on this thread.

**Reply by dm30** [reply]
Could it be Ente links? Or albums?

**Reply by vishnukvmd** [reply]
Links would work!

**Reply by dm30** [reply]
Appreciated Vishnu. I have loads of pics and I don't know what is better for Ente in this endeavour. Anyway, I limited myself to just some simple DNGs and few more Apple's ProRAW format, which are still standardised DNG and open-source, although a little different. Unless otherwise stated and granted the mobile orientation of Ente (currently) I think its better to handle this mobile formats, which cover the vast majority of smartphones, before delving in camera-maker specific ones. Even though my Xiaomi phone shoots 'Ultra RAW' variety and there's still a couple more of proprietary formats around. You'll have notice I added you to 'DNG' and 'ProRAW' albums. Thanks in advance

**Reply by dm30** [reply]
Is that right Vishnu?

**Comment by slueppken** [comment]
It seems all RAW files generated by the app ProShot on Android (https://www.riseupgames.com/proshot) have black previews in Ente and appear as black and white when opening them. Hope that helps.

**Comment by milesfoxy** [comment]
Adding to this request, I just signed up for Ente but if I am going to get a new phone this year to do RAW or HEIF DNG formats, I want this service to support what I will primarily use.

**Comment by elRicharde** [comment]
What about Fuji RAF? 
I can´t even upload these files :-(

**Reply by mupster** [reply]
I shared the RAF, NEF and DNG samples at the start of this thread which Vishnu shared an album link to

https://albums.ente.io/?t=4edQY8Lv#JBdFEj4oYbYZgerNAsnN2qVyBS174PvZ39iUtpJYKNWQ

I'm guessing it's a case of prioritising features, hopefully with the recent increased interest this feature might get bumped up the list.

**Comment by dm30** [comment]
On a second thought, I think RAW is not that essential right now if because this files will pollute the gallery sited often along JPG pair. So while support is being implemented, a RAW filter option for viewing your files will be very useful. I mean, to un-check RAW images.

**Comment by dm30** [comment]
Thank you

Enviado con el correo electrónico seguro de [Proton Mail](https://pr.tn/ref/XY7Q8GA046QG).

**Comment by mlkr4** [comment]
Still no support for TIFF images, as in uncompressed scans from film scanners, example: https://albums.ente.io/?t=dFVCUHcF#9ufQmBSVPzReSCnu7fdZ3oUnk69RaLPvxd4vsR6smeFW

**Comment by titivermeesch** [comment]
It has been mentioned a few times here already, but would love support for CR2 and CR3 coming from my Canon cameras

**Reply by TranceCat42** [reply]
Me too! My .cr3's wouldn't even upload. Seeing them in thumbnail is one thing but not being able to back them up is a problem.

**Reply by titivermeesch** [reply]
> Me too! My .cr3's wouldn't even upload. Seeing them in thumbnail is one thing but not being able to back them up is a problem.

I managed to upload them through the desktop app, but on mobile it wouldn't upload at all. (No previews on desktop app either)

**Comment by Bielecki** [comment]
What is the status of this feature? Help page says
> we only have limited support for RAW currently

Which RAW formats Ente currently supports?

**Comment by v1nodp** [comment]
As I said before, CR2 is not supported. The pictures/thumbnails are all black. But here's something strange: when we use faces, some CR2 pictures (black ones) also show up. I'm not sure if this is a bug or if the face recognition software is recognizing the face in an unsupported format.

**Comment by Facecloth4523** [comment]
Just checking in on this one

**Comment by ferd-i** [comment]
Any update?

**Comment by buster** [comment]
I am planning to migrate around 300GB pictures and movies to Ente and there is a significant amount of ORF files in there.
I'm also interested in some updates.

**Reply by Bielecki** [reply]
I got tired of waiting and I'm using Immich. They even released a stable version lately. Not sure if they support ORF, but DNG is handled just fine.

**Reply by titivermeesch** [reply]
I really love Ente so I still want to wait, but it's indeed a critical feature that's preventing me from fully enjoying it. I'm shooting RAW+JPG just because I want to see the previews on my phone

**Comment by v1nodp** [comment]
The strange thing is that face recognition works with my RAW files (like .CR2), but the thumbnails appear blank. When I search for a person’s photo, the RAW files show up through face recognition, yet their thumbnails are all black

**Reply by juliadin** [reply]
I seem to still have this problem at least in the desktop app... need to check with the mobile app. I am pretty sure (though not certain) that I have seen SOME cr2 raws with thumbnails... are they generated over time maybe? idk.

**Comment by konradstrg** [comment]
.CR3 photos still not generating thumbnails

**Comment by PhiPa87** [comment]
Any Update on this topic or do i really have to use now a different app? Seeing only black thumbnails makes it not usable when more then half of my pictures are RAW files (.CR2 and .DNG)

**Comment by gweiermann** [comment]
I’d be really interested in a feature that groups a JPG and RAW file together when both exist for the same photo.

In that case, the JPG already provides a perfectly good preview, so there’s no real need to show the RAW as a separate gallery item. When downloading, Ente could simply ask whether to download the JPG, the RAW, or both.

I think this would make shooting in RAW+JPG much nicer to use in Ente, while still keeping both original files available.

I drafted a quick AI-generated mockup of what this could look like:
<img width="1448" height="1086" alt="image" src="https://github.com/user-attachments/assets/0fbcbd11-2abc-4212-82fe-9989c1e7ad13" />

**Reply by t-m-w** [reply]
Yeah, the Google Photos app does it nicely.
